runtime/vm/createramclass.cpp

```diff
@@ -2093,8 +2109,21 @@ checkFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *clas
 static J9Class*
 internalCreateRAMClassDropAndReturn(J9VMThread *vmThread, J9ROMClass *romClass, J9CreateRAMClassState *state)
 {
-	/* pop protectionDomain */
-	DROP_OBJECT_IN_SPECIAL_FRAME(vmThread);
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+	/* This should only done once per ROM class. If two array ram
+	 * classes are being created skip the first one. state->ramClass
+	 * will hold the array class at this point.
+	 */
+	if (FALSE == ((NULL != state->ramClass) &&
+			J9CLASS_IS_ARRAY(state->ramClass) &&
+			J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(((J9ArrayClass*)state->ramClass)->componentType) &&
+			J9_ARE_NO_BITS_SET(state->valueTypeFlags, J9ClassArrayIsNullRestricted)
+		))
```

Comment:
The `&&` should be at the start of the line, and the enclosing `)` should be inline with the `if`.
```
if (FALSE == ((NULL != state->ramClass)
	        && J9CLASS_IS_ARRAY(state->ramClass)
	        && J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(((J9ArrayClass*)state->ramClass)->componentType)
	        && J9_ARE_NO_BITS_SET(state->valueTypeFlags, J9ClassArrayIsNullRestricted))
)
```